tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -28,7 +28,29 @@ use tracing_core::{
 
 /// A `Layer` which filters spans and events based on a set of filter
 /// directives.
-// TODO(eliza): document filter directive syntax?
+///
+/// # Directives
+///
+/// Directives are a set of instructions on how to filter tracing events. The directive
+/// syntax is similar to the one presented in `env_logger`. The syntax consists mainly
+/// of four parts `target[span{field=value}]=level`.
+///
+/// - `target` matches on the target that the event comes from, generally this will be the
```

Comment:
This should link to where we define targets in the `tracing` docs: https://docs.rs/tracing/0.1.10/tracing/struct.Metadata.html